helios-services/src/main/java/com/spotify/helios/rollingupdate/DeploymentGroupEventFactory.java

```diff
@@ -37,51 +35,58 @@
-                                                        final RolloutTask task) {
+  private Map<String, Object> createEvent(final String eventType,
+                                          final DeploymentGroup deploymentGroup) {
     final Map<String, Object> ev = Maps.newHashMap();
-    ev.put("eventType", "rollingUpdateTaskSucceeded");
+    ev.put("eventType", eventType);
     ev.put("timestamp", System.currentTimeMillis());
-    ev.put("deploymentGroupName", deploymentGroup.getName());
+    ev.put("deploymentGroup", deploymentGroup);
+    return ev;
+  }
+
+  private Map<String, Object> addTaskFields(final Map<String, Object> ev,
+                                            final RolloutTask task) {
     ev.put("action", task.getAction());
     ev.put("target", task.getTarget());
-    ev.put("result", RolloutTask.Status.OK);
     return ev;
   }
 
+  public Map<String, Object> rollingUpdateTaskFailed(final DeploymentGroup deploymentGroup,
+                                                     final RolloutTask task,
+                                                     final String error,
+                                                     final RollingUpdateError errorCode) {
+    final Map<String, Object> ev = createEvent("rollingUpdateTaskResult", deploymentGroup);
+    ev.put("success", 0);
+    ev.put("error", error);
+    ev.put("errorCode", errorCode);
+    return addTaskFields(ev, task);
+  }
+
+  public Map<String, Object> rollingUpdateTaskSucceeded(final DeploymentGroup deploymentGroup,
+                                                        final RolloutTask task) {
+    final Map<String, Object> ev = createEvent("rollingUpdateTaskResult", deploymentGroup);
+    ev.put("success", 1);
+    return addTaskFields(ev, task);
+  }
+
   public Map<String, Object> rollingUpdateStarted(final DeploymentGroup deploymentGroup,
-                                                  final RollingUpdateReason reason,
-                                                  final JobId jobId) {
-    final Map<String, Object> ev = Maps.newHashMap();
-    ev.put("eventType", "rollingUpdateStarted");
-    ev.put("timestamp", System.currentTimeMillis());
-    ev.put("deploymentGroupName", deploymentGroup.getName());
+                                                  final RollingUpdateReason reason) {
+    final Map<String, Object> ev = createEvent("rollingUpdateStarted", deploymentGroup);
     ev.put("reason", reason);
-    ev.put("jobId", jobId.toString());
     return ev;
   }
 
   public Map<String, Object> rollingUpdateDone(final DeploymentGroup deploymentGroup) {
-    final Map<String, Object> ev = Maps.newHashMap();
-    ev.put("eventType", "rollingUpdateDone");
-    ev.put("timestamp", System.currentTimeMillis());
-    ev.put("deploymentGroupName", deploymentGroup.getName());
-    ev.put("state", DeploymentGroupStatus.State.DONE);
+    final Map<String, Object> ev = createEvent("rollingUpdateFinished", deploymentGroup);
```

Comment:
No. The eventType is rollingUpdatedFinished both for rollingUpdateDone and rollingUpdateFailed